x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/watch/WatchParser.java

```diff
@@ -174,6 +175,13 @@ public Watch parse(String id, boolean includeStatus, WatcherXContentParser parse
                 throw new ElasticsearchParseException("could not parse watch [{}]. unexpected field [{}]", id, currentFieldName);
             }
         }
+        /* Make sure we are at the end of the available input data -- certain types of JSON errors will not manifest
+             until we try to consume additional tokens.
+        */
+        if (parser.nextToken() != null) {
+            throw new ElasticsearchParseException("could not parse watch [{}].  unexpected data beyond [line: {}, column: {}]",
```

Comment:
minor nit: maybe we can come up with a more explaining error message here, as `unexpected data` can mean anything - however I don't have a smarter idea as well to be honest. Is it `unparseable`/`unknown` or just `too much data`/`unneeded additional data`?

Reply:
@spinscale Actually it is this code that enables the JsonParseException for the original test case.  Ryan asked that the code for the test case be changed, which resulted in a different error path.  However, the key factor is that we now attempt to grab the next token beyond what we believe to be the last token of the watch definition.  A previous revision of this code made that fact clear, at the expense of some more complex code.

Reply:
yea I think what @spinscale is saying is that he would like the error message changed.  Maybe `expected end of payload/data/json but received additional data`

Reply:
@hub-cap I was trying to respond to @spinscale's comment above in the LGTM -- I'll rework the error message